backend/src/main/java/ai/giskard/service/InspectionService.java

```diff
@@ -167,6 +167,16 @@ private Selection getSelection(Inspection inspection, Filter filter) throws File
                     DoubleColumn absDiff = calculatedTable.doubleColumn("absDiff");
                     selection = correctSelection.and(absDiff.isLessThanOrEqualTo(applicationProperties.getBorderLineThreshold()));
                 }
+                case OVERCONFIDENCE -> {
+                    DoubleColumn overconfidenceScore = calculatedTable.doubleColumn("absDiff");
+                    DoubleColumn overconfidenceThreshold = DoubleColumn.create("overconfidenceThreshold", overconfidenceScore.size());
+                    for (int i = 0; i < overconfidenceScore.size(); i++) {
+                        int n = predsTable.columnCount();
```

Comment:
If the threshold is the same value for every row, this should be computed outside of the loop.